control/config/fabrid.go

```diff
@@ -0,0 +1,136 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"io"
+	"net"
+	"strings"
+
+	"github.com/scionproto/scion/pkg/private/serrors"
+	"github.com/scionproto/scion/pkg/segment/extensions/fabrid"
+	"github.com/scionproto/scion/private/config"
+)
+
+type FABRIDPolicy struct {
+	IsLocalPolicy    bool                     `yaml:"local,omitempty"`
+	LocalIdentifier  uint32                   `yaml:"local_identifier,omitempty"`
+	LocalDescription string                   `yaml:"local_description,omitempty"`
+	GlobalIdentifier uint32                   `yaml:"global_identifier,omitempty"`
+	SupportedBy      []FABRIDConnectionPoints `yaml:"connections,omitempty"`
+}
+
+// Validate validates that all values are parsable.
+func (cfg *FABRIDPolicy) Validate(asInterfaceIDs []uint16) error {
+	for _, connectionPoint := range cfg.SupportedBy {
+		connectionPoint.asInterfaceIDs = asInterfaceIDs
+		if err := config.ValidateAll(&connectionPoint); err != nil {
+			return serrors.WrapStr("Validating supported interfaces failed", err)
+		}
+		connectionPoint.asInterfaceIDs = nil
+	}
+	if cfg.IsLocalPolicy && (cfg.LocalIdentifier == 0 || cfg.LocalDescription == "") {
+		return serrors.New("Local policy configuration must not be empty.")
+	} else if !cfg.IsLocalPolicy && cfg.GlobalIdentifier == 0 {
+		return serrors.New("Global policy identifier must be valid.")
+	}
+
+	return nil
+}
+
+// Sample writes a config sample to the writer.
+func (cfg *FABRIDPolicy) Sample(dst io.Writer, path config.Path, ctx config.CtxMap) {
+	config.WriteString(dst, fabridLocalPolicySample)
+}
+
+type FABRIDConnectionPoints struct {
+	Ingress   FABRIDConnectionPoint `yaml:"ingress,omitempty"`
+	Egress    FABRIDConnectionPoint `yaml:"egress,omitempty"`
+	MPLSLabel uint32                `yaml:"mpls_label,omitempty"`
+	// asInterfaceIDs is manually set to verify that both the provided
+	// ingress and egress actually belong to valid AS interfaces if their
+	// type is interface
+	asInterfaceIDs []uint16
+}
+
+// Validate validates that all values are parsable.
+func (cfg *FABRIDConnectionPoints) Validate() error {
+	doInterfacesExist := func(connectionPoints ...*FABRIDConnectionPoint) bool {
+		for _, cp := range connectionPoints {
+			if cp.Type == fabrid.Interface {
+				found := cp.Interface == 0
+				for i := 0; i < len(cfg.asInterfaceIDs); i++ {
+					if cp.Interface == cfg.asInterfaceIDs[i] {
+						found = true
+						break
+					}
+				}
+				if !found {
+					return false
+				}
+			}
+		}
+		return true
+	}
+	if cfg.Ingress.Type != fabrid.Interface && cfg.Ingress.Type != fabrid.Wildcard {
+		return serrors.New("FABRID policies are only supported from an interface to an IP" +
+			" range or other interface.")
+	} else if cfg.Ingress.Type == fabrid.Interface && cfg.Ingress.Interface == 0 {
```

Comment:
Move down to FABRIDConnectionPoint, also check for inverse (type!=interface, interface != 0)